### PR TITLE
simplify and add aaad api

### PR DESCRIPTION
**Pull request by ankanm**
ROGER-1358 and ROGER-1359.
- [x] Create a more standard python project. 
- [x] Moving to flask.
- [x] Added UI + better handling of auth etc.
- [x] Simpler/modularized/standard code.
- [x] Cookie/session handling more standard (and more secure than the previous one).
- [x] Update ansible scripts for new setup.
- [x] Change how aaad is started on the server.
- [x] Ensure logging is appropriate.
- [x] Updates to nginx (and test).
- [x] Remove/deprecate old `aaad.py`

This is ~~still WIP~~ mostly complete and ready for review.

fyi - @amitbose327 @vmahedia

### CODE REVIEW

File: aaad/aaad_app.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    Because of the @login_required decorator authentication should be already complete
+    using one of the supported methods by the time we reach this method. So, let's just
+    worry about authorizaton.
+    '''
+    user = current_user.get_username()
+    actas = request.cookies.get("actas") # try to get it from cookie
+    if not actas:
+        actas = request.headers.get("act_as_user") # try to get from header
+    if not actas:
+        actas = user # default to the user
```
Comment (vmahedia) [inline aaad/aaad_app.py]: In coolies The key is "actas" here it's "act_as" should not both be same

File: aaad/aaad_app.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    This method filters the response (body) based on the access the current user has.
+    Should typically be called by the proxy (internal) and not from outside.
+    '''
+    user = current_user.get_username()
+    actas = request.cookies.get("actas") # try to get it from cookie
+    if not actas:
+        actas = request.headers.get("act_as_user") # try to get from header
+    if not actas:
+        actas = user # default to the user
+
```
Comment (vmahedia) [inline aaad/aaad_app.py]: Why not make a function, code duplication eree

File: aaad/aaad.py [hunk 3]
```diff
@@ -21,7 +21,6 @@
 secret_key = os.getenv('SECRET_KEY', 'test_key_NOT_a_secret')
 session_id_key = os.getenv('SESSION_ID_KEY', 'aaadsid')
 session_id_domain = os.getenv('SESSION_ID_DOMAIN', None)
-htpasswd_file = os.getenv('HTPASSWD_FILE', '')
 
 class AuthHTTPServer(ThreadingMixIn, HTTPServer, ):
     pass
```
Comment (miroswan) [inline aaad/aaad.py]: There is a trailing comma in the class parameter list.

File: aaad/aaad_app.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+import os
+import json
+import logging
+
+from authenticators import FileAuthenticator
+from authorizers import FileAuthorizer
+from resources import Users, Groups, CanActAsUsers
+from sessions import login_manager, SessionUser
+import sessions
+
```
Comment (miroswan) [inline aaad/aaad_app.py]: There is a redundant import here. Also, I believe it's best practice to prepend each imported module with a dot character when doing local imports. 

Reference: https://docs.python.org/2.5/whatsnew/pep-328.html

File: aaad/aaad_app.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    This method finds out if the current user is authorized to -
+        - perform the action specified in 'method' header
+        - on the resource specified in 'URI header'
+    The request body and content_type are also required for ths purpose.
+
+    Note:
+    Because of the @login_required decorator authentication should be already complete
+    using one of the supported methods by the time we reach this method. So, let's just
+    worry about authorizaton.
+    '''
```
Comment (miroswan) [inline aaad/aaad_app.py]: Nice docs!

File: aaad/authorizers.py [hunk 6]
```diff
@@ -127,9 +131,73 @@ def authorize(self, user, act_as, resource, logging, info, data, content_type, a
...
+            allowed_users_list = []
+            self._get_merged_data(act_as, allowed_users_list, [], permissions, '')
+            allowed_namespace_patterns = []
+            for user in allowed_users_list:
+                if 'allowed_names' in permissions[user]:
+                    for pattern in permissions[user]['allowed_names']:
+                        if not pattern in allowed_namespace_patterns:
+                            allowed_namespace_patterns.append(pattern)
+            return allowed_namespace_patterns
+
```
Comment (miroswan) [inline aaad/authorizers.py]: Nice small methods. Smaller methods are are easier to unit test than bigger ones.

File: aaad/templates/index.html [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        border-color: #000;
+      }
+      textarea {
+        vertical-align: top;
+        height: 5em;
+        resize: vertical;
+      }
+      .button {
+        font: 1em sans-serif;
+      }
```
Comment (miroswan) [inline aaad/templates/index.html]: Could be consider breaking out the CSS?

File: aaad/aaad.py (hunk 3 above)
Comment (ankanm) [inline aaad/aaad.py]: The new code replaces this whole module. 
I'll just replace this file with the new one (aaad_app.)

File: aaad/aaad_app.py (hunk 1 above)
Comment (ankanm) [inline aaad/aaad_app.py]: They don't have to be. I wanted to use a smaller name for the cookies. But didn't change the header. 
I'm hoping we'll stop using the basic auth + this header and switch over to cookie/token based auth entirely.

File: aaad/templates/index.html (hunk 7 above)
Comment (ankanm) [inline aaad/templates/index.html]: Yes! It was not working with the styles being outside. Turns out it was an nginx configuration for the mime type.

**Pull request merged**